src/main/java/org/mariadb/jdbc/internal/stream/PacketOutputStream.java

```diff
@@ -255,66 +225,75 @@ private void checkPacketMaxSize(int limit) throws MaxAllowedPacketException {
     }
 
     private void internalFlush() throws IOException {
-        buffer.flip();
-        int limit = buffer.limit();
-        if (limit > 0) {
-            checkPacketMaxSize(limit);
-
-            byte[] bufferBytes = new byte[0];
-            int notCompressPosition = 0;
-            int expectedPacketSize = limit + HEADER_LENGTH * ((limit / maxPacketSize) + 1);
-            if (useCompression) {
-                bufferBytes = new byte[expectedPacketSize];
-            }
-
-            while (notCompressPosition < expectedPacketSize) {
-                int length = buffer.remaining();
-                if (length > maxPacketSize) {
-                    length = maxPacketSize;
-                }
-                if (useCompression) {
-                    bufferBytes[notCompressPosition++] = (byte) (length & 0xff);
-                    bufferBytes[notCompressPosition++] = (byte) (length >>> 8);
-                    bufferBytes[notCompressPosition++] = (byte) (length >>> 16);
-                    bufferBytes[notCompressPosition++] = (byte) seqNo++;
-                } else {
-                    byte[] header = new byte[HEADER_LENGTH];
-                    header[0] = (byte) (length & 0xff);
-                    header[1] = (byte) (length >>> 8);
-                    header[2] = (byte) (length >>> 16);
-                    header[3] = (byte) seqNo++;
-                    outputStream.write(header);
-                    notCompressPosition += 4;
-                }
-
-                if (length > 0) {
-                    if (useCompression) {
-                        buffer.get(bufferBytes, notCompressPosition, length);
-                        notCompressPosition += length;
-                    } else {
-                        if (buffer.hasArray()) {
-                            outputStream.write(buffer.array(), buffer.position(), length);
-                            buffer.position(buffer.position() + length);
-                        } else {
-                            bufferBytes = new byte[length];
-                            buffer.get(bufferBytes, 0, length);
-                            outputStream.write(bufferBytes, 0, length);
-                        }
-                        notCompressPosition += length;
-                        outputStream.flush();
-                    }
-                }
-            }
-
-
-            if (useCompression) {
-                //now bufferBytes in filled with uncompressed data
-                compressedAndSend(notCompressPosition, bufferBytes);
-            }
-        }
-
-    }
-
+    	buffer.prepare();
```

Comment:
I am not sure if maria has published one.  But it's very close to the style for my project:

https://github.com/threadly/threadly/blob/master/config/Eclipse_Coding_Style.xml

The biggest change is threadly uses 2 spaces, and mariaDb uses 4.

FWIW, I don't feel every detail of style has to match exactly (others may, not sure).  But the choice to use tabs instead of 4 spaces (or to use a different quantity of spaces) can be the most disruptive visually.